src/chapters/interfaces-and-polymorphism/new-studio.rst

```diff
@@ -0,0 +1,184 @@
+abstract class vs. each specific class.
+
+Just like you did with your Restaurant Menu studios, you will begin with pen
+and paper (or similar tools).
+
+Consider Generic Behaviors
+---------------------------
+
+Here are a few behaviors that both CDs and DVDs share:
+
+a. They both spin,
+b. They both store data,
+c. Data gets written to the discs by using a laser,
+d. The data on the discs can be read by using a laser,
+e. Once loaded, they both report information like name, capacity, contents, and
+   disc type.
+
+With your partner, add one or two more behaviors to the list. Feel free to
+do some quick research online if needed. Note that information like storage
+capacity does NOT belong on this list, since it describes a *property*
+(field) rather than a *behavior* (method).
+
+For each behavior, identify if it depends on any type of instance variable.
+For example, the spinning behavior does not require any field information
+from the ``CD`` or ``DVD`` classes.
+
+This collecting and sorting of behaviors is the first step toward designing
+your interface and abstract class. It prompts you to consider behaviors that
+can be generally applied vs. those that depend on data stored in a specific
+object.
+
+Consider Class-Specific Behavior and Data
+------------------------------------------
+
+Next, you and your partner need to decide which behaviors go into the
+interface vs. an abstract class.
+
+Behaviors added to an interface do NOT have to be identical for every object
+that implements the interface. The key is that the methods require no
+object-specific data. The methods can accept parameters, but these should
+represent generic values like ``String searchTerm`` rather than field data
+like ``DVD.title``.
+
+For example, even though CDs and DVDs both spin, they do so at different
+speeds. Their *behavior* (spinning) is the same, but the details of that
+behavior vary. Thus, a method like ``spinDisc()`` fits well into an interface.
+
+#. For the other behaviors you and your partner sorted, decide which ones
+   belong in the interface.
+#. Write down details about how these behaviors differ for CDs and DVDs. This
+   guides your thinking about how each class should override the methods.
+#. For the behaviors that cannot go into the interface, decide if they could be
+   placed in an abstract class or if they need to appear in the separate ``CD``
+   and ``DVD`` classes.
+#. Finally, write down the fields needed to define the properties of a CD or
+   DVD (e.g. ``storageCapacity``). Consider:
+
+   a. Which fields are shared by both types of discs, and which ones are
+      unique to each class?
+   b. Do any of the fields represent *constants*?
+
+Code the Interface
+-------------------
+
+Now that your team has completed the before-coding thinking, it's time to
+start building!
+
+#. Create an ``OpticalDisc`` file for the interface. Refer back to
+   (*TODO: Add relevant chapter link*) if you need a quick review of how to do
```

Comment:
Add link!

Reply:
Now that lesson 7 is merged into master... done!